crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -156,7 +156,24 @@ impl Ethereum {
             access_list: Some(tx.access_list.into()),
             // Specifically set high gas because some nodes don't pick a sensible value if omitted.
             // And since we are only interested in access lists a very high value is fine.
-            gas: Some(self.block_gas_limit().0),
+            gas: Some(match self.inner.chain {
+                // Arbitrum has an exceptionally high block gas limit (1,125,899,906,842,624),
+                // making it unsuitable for this use case. To address this, we use a
+                // fixed gas limit of 100,000,000, which is sufficient
+                // for all solution types, while avoiding the "insufficient funds for gas * price +
+                // value" error that could occur when a large amount of ETH is
+                // needed to simulate the transaction, due to high transaction gas limit.
+                //
+                // If a new network is added, ensure its block gas limit is checked and handled
+                // appropriately to maintain compatibility with this logic.
+                Chain::ArbitrumOne => 100_000_000.into(),
```

Comment:
Does it make sense to make it an optional config that can be enabled on any network to avoid this matching?

Reply:
Would be ok to have, but I am not sure we should put every single thing into config, otherwise config will soon be huge. This is a pretty small implementation detail and probably won't change once set.

Reply:
If we are 100% certain about the hardcoded value, let's keep it as it is.